app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/GitUtils.java

```diff
@@ -20,14 +20,27 @@ public class GitUtils {
     public static final Duration RETRY_DELAY = Duration.ofSeconds(1);
     public static final Integer MAX_RETRIES = 20;
 
+    /**
+     * Pattern for validating the ssh address if that starts with a scheme
+     * Should start with ssh://
+     * may or may not have a username : e.g. ssh://domain.xy:/path/path.git
+     * username can start with any small alphabet or a _ underscore
+     * the RFC host name should have regex : ((?!-)[A-Za-z0-9-]{1,63}(?<!-)\\.)+[A-Za-z0-9-]{1,63}\\.[A-Za-z]{2,6},
+     * however due to already existing leniency
+     * hostname can have all alphanumerics, -, and .  e.g. ssh://_ab-xy@ab-12.ab:/v3/newJet/ai/zilla.git
+     * the port number could be not present as well. e.g. ssh://_ab-xy@domain.com:/v3/newJet/ai/zilla
+     */
     public static final Pattern URL_PATTERN_WITH_SCHEME =
-            Pattern.compile("^ssh://git@(?<host>.+?)(:(?<port>\\d+))?/+(?<path>.+?)(\\.git)?$");
+            Pattern.compile("^ssh://([a-z_][\\w-]+@)?(?<host>[\\w-.]+?)(:(?<port>\\d*))?/+(?<path>.+?)(\\.git)?$");
```

Comment:
Since the host matcher is a lot more specific now (instead of a `.`), we can do greedy match here.

```suggestion
            Pattern.compile("^ssh://([a-z_][\\w-]+@)?(?<host>[\\w-.]+)(:(?<port>\\d*))?/+(?<path>.+?)(\\.git)?$");
```